Kernel/FileSystem/Ext2FileSystem.cpp

```diff
@@ -426,6 +426,10 @@ KResult Ext2FSInode::flush_block_list()
         ++output_block_index;
         --remaining_blocks;
     }
+    // Unused direct block pointers should be zeroed.
```

Comment:
I think this would be more valuable as a **why** comment, instead it's current form of a **what** comment that re-states what the code does in words.

Reply:
In my opinion this already is a why comment. It tells you the reason these values are being set to zero because the pointers in them are **unused**, which is otherwise not immediately clear from the context.

You could of course go further by explaining why unused pointers should be zeroed. Aside from the fact that fsck complains if they are not zeroed I don't know why they have to be.

Reply:
I guess we could add a note about `e2fsck` complaining if these are not zeroed out.